src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,245 @@
+                      segment: ISegment
+  ): Promise<unknown> => {
+    const killJob$ = new ReplaySubject();
+
+    const abortError$ = killJob$.pipe(
+      map(() => {
+        throw new VideoThumbnailLoaderError("ABORTED",
+                                            "VideoThumbnailLoaderError: Aborted job.");
+      })
+    );
+
+    const jobPromise = observableMerge(
+      initSourceBuffer$(contentInfos,
+                        this._videoElement).pipe(
+        mergeMap((videoSourceBuffer) => {
+          const removeBuffers$: Observable<unknown> =
+            removeBufferForTime$(this._videoElement, videoSourceBuffer, time);
+          return removeBuffers$.pipe(
+            mergeMap(() => {
+              log.debug("VTL: Removed buffer before appending segments.", time);
+
+              return segmentLoader({
+                manifest: contentInfos.manifest,
+                period: contentInfos.manifest.periods[0],
+                adaptation: contentInfos.adaptation,
+                representation: contentInfos.representation,
+                segment,
+              }).pipe(
+                filter((evt): evt is { type: "data";
+                                       value: { responseData: Uint8Array }; } =>
+                  evt.type === "data"),
+                mergeMap((evt) => {
+                  const inventoryInfos = { manifest: contentInfos.manifest,
+                                           period: contentInfos.period,
+                                           adaptation: contentInfos.adaptation,
+                                           representation: contentInfos.representation,
+                                           segment };
+                  return videoSourceBuffer
+                    .pushChunk({ data: { chunk: evt.value.responseData,
+                                         timestampOffset: 0,
+                                         appendWindow: [undefined, undefined],
+                                         initSegment: null,
+                                         codec: contentInfos
+                                           .representation.getMimeTypeString() },
+                                 inventoryInfos })
+                      .pipe(map(() => {
+                        log.debug("VTL: Appended segment.", evt.value.responseData);
+                        this._videoElement.currentTime = time;
+                        if (this._videoElement.buffered.length === 0) {
+                          throw new VideoThumbnailLoaderError("NOT_BUFFERED",
+                                                              "No buffered data after loading.");
+                        } else {
+                          return time;
+                        }
+                      })
+                    );
+                })
+              );
+            })
+          );
+        }),
+        catchError((err: Error | { message?: string; toString(): string }) => {
```

Comment:
Why not just unknown then duck typing?

Reply:
Not sure about this. Duck typing with an `unknown` type requires to cast the error (`as IType`) to check if it has a property.

Reply:
Yeah but here we use an any ([hidden in the catchError declaration](https://github.com/ReactiveX/rxjs/blob/master/src/internal/operators/catchError.ts#L107)). That's worse to me.

Reply:
I still find it more safe to use an `unknown` then duck typing then a type cast with `as` than relying on `catchError` being `any` and crossing fingers.

Reply:
Still the case for me